core/java/com/google/instrumentation/stats/View.java

```diff
@@ -32,15 +32,48 @@
   /**
    * Applies the given match function to the underlying data type.
    */
-  public abstract <T> T match(
-      Function<DistributionView, T> p0,
-      Function<IntervalView, T> p1);
+  public abstract <T> T match(Visitor<T> visitor);
 
   // Prevents this class from being subclassed anywhere else.
   private View() {
   }
 
   /**
+   * Visitor for the View type.
+   */
+  public abstract static class Visitor<T> {
+    public abstract T apply(DistributionView view);
+
+    public abstract T apply(IntervalView view);
+
+    /**
+     * Creates a Visitor with two {@link Function} objects.
+     *
+     * <p>Example:
```

Comment:
eventhough the local name says what this does, I'd still spell it out in the Example text.

ex. `<p>For example, the following visitor counts the aggregations in a view:`